### ISSUE TITLE
Use a neutral format to have lossless interface with JSON, Xarray, scipp, pandas

### ISSUE THREAD

**Issue by loco-philippe** [astropy/astropy#16286]
### What is the problem this feature will solve?

Each tool has a specific structure for processing multidimensional data with the following consequences:

- interfaces dedicated to each tool,
- partially processed data,
- no unified representation of data structures

### Describe the desired outcome

The proposed format  (see [jupyter notebook](https://nbviewer.org/github/loco-philippe/ntv-numpy/blob/main/example/example_ntv_numpy.ipynb), [github repository](https://github.com/loco-philippe/ntv-numpy/blob/main/README.md), [PyPI package](https://pypi.org/project/ntv-numpy/) ) is based on the following principles:

- neutral format available for tabular or multidimensional tools (e.g. Numpy, pandas, xarray, scipp, astropy),
- taking into account a wide variety of data types as defined in [NTV](https://www.ietf.org/archive/id/draft-thomy-json-ntv-02.html) format,
- high interoperability: reversible (lossless round-trip) interface with tabular or multidimensional tools,
- reversible and compact JSON format,
- Ease of sharing and exchanging multidimensional and tabular data,

### Additional context

https://github.com/numpy/numpy/issues/12481#issuecomment-2049179803

**Comment by pllim** [comment]
Based on the linked comment, looks like you specifically is targetting `NDData` only? 

Is this related to the "Array API" stuff?

Hope you can clarify. Thanks!

**Comment by loco-philippe** [comment]
Thank you @pllim for your comment,

Considering Astropy, yes, `NDData` is well targeted (and `table` in the next version).

I'm not sure I understand the "Array API" stuff, if you think about [the Python Array API standard](https://data-apis.org/array-api/latest/), this initiative is not still included (but I think it should be).

**Comment by pllim** [comment]
Yes, that Array API. I just do not want this to be in conflict with the following issue, that might or might not be related:

* #15073
